prow/scripts/cluster-integration/helpers/deprovision-gardener-cluster.sh

```diff
@@ -15,4 +15,4 @@
 readonly NAMESPACE="garden-${GARDENER_PROJECT_NAME}"
 
 kubectl --kubeconfig "${GARDENER_CREDENTIALS}" -n "${NAMESPACE}" annotate shoot "${GARDENER_CLUSTER_NAME}" confirmation.garden.sapcloud.io/deletion=true --overwrite
-kubectl --kubeconfig "${GARDENER_CREDENTIALS}" -n "${NAMESPACE}" delete shoot "${GARDENER_CLUSTER_NAME}" --wait=false
```

Comment:
Is there a reason to wait for deletion?

We added the `--wait=false` flag so that jobs would finish right away. The problem is that on gardener, cluster deletion often goes wrong and then the cluster can stay for up to 5 days until it is garbage collected.
In those cases, the job would timeout and fail even if all went well.